### ISSUE TITLE
Add an option to scrollToAlignment that will minimize scroll events when moving from each row to the next

### ISSUE THREAD

**Issue by adamtal3** [bvaughn/react-virtualized#455]
Thanks for your great effort with this project!

### TL;DR - What I want to accomplish

I want the table scroll position to update as less as possible - **scroll the item to the top only if outside of view area**.

### Further explanation of my specific use case

I'm using a `<FlexTable />` to show a big list when each item of the list requires some time to render (not too much but it does have an impact on the user experience).

The selected item should always be visible, I had `scrollToAlignment="center"` and `scrollToIndex={currentIndex}` (currentIndex updates every few seconds to the next index - I'm displaying a list of captions for a playing video).

I tried removing `scrollToAlignment="center"` hoping to get less renders on each update, but then, when the index reaches the bottom of the visible rows it scroll the same amount on each update as it did when it was set to `center` (it scrolls `rowHeight` pixels).

I was wondering if it would be possible to minimize the scrolling by moving the current row to the top when reaching the bottom and then not scrolling until reaching to the bottom again. That way the number of re-renders (in my case, when moving from one row to the next every few seconds) will be minimized.

Is there's any way to get this kind of functionality from `react-virtualized` today? Do you think I have a better option then minimizing the scrolling events? or do you think it should be a new feature?

Thanks!

**Comment by bvaughn** [comment]
Hey @adamtal3,

Thanks for the lengthy explanation of what you're looking for.

Pretty sure what you're describing you want is the default behavior, (`scrollToAlignment="auto"`). It ensures that the specified index is always _fully_ visible, but with the least amount of scrolling possible (including no scrolling at all if it is already visible).

You can test this behavior [here](https://bvaughn.github.io/react-virtualized/#/components/Table). In the "scroll to index" field, type any number between 0-5 and the table won't scroll at all. Type 6 and it will scroll exactly 1 row to ensure that the row is fully visible.

I don't think I can add more control over this functionality than I already have without complicating the Api (which I don't want to do). If you have specific proposals though- I'll be happy to hear them. Also, a Plnkr or some example I can look at that shows what you're referring to is usually helpful. 😄

I think this is not an actionable item for me so I'm going to close it for now- but we can continue talking on the ticket. I'll still see and respond to updates.

**Comment by asakaplan** [comment]
I hope this isn't outdated at this point, but I was hoping to reopen this with a specific proposal: that "auto" is treated as a modifier rather than be its own option.

Reason:
Ctrl+F search in a browser works by scrolling so that a search is in the center if the window needed to scroll, otherwise it doesn't scroll at all. This could be `auto center`.

Similarly, `auto start`/`auto end` would scroll to the top/bottom only if it needs to scroll.

If you agree with the proposal, I could submit a pull request.

**Comment by bvaughn** [comment]
I would review a PR and consider it, if it didn't add much complexity or otherwise harm performance. Sure!
